src/transactions.rs

```diff
@@ -411,7 +457,7 @@ impl<L: Ledger + Serialize + DeserializeOwned> Transactions<L> {
         params: TransactionParams<L>,
     ) -> Result<TransactionEditor<'_, L>, KeystoreError<L>> {
         let txn = Transaction::<L> {
-            uid: params.uid,
+            uid: params.uid.unwrap(),
```

Comment:
Not a huge fan of this `unwrap`. For required params, I think it would be better if we could take them as parameters of `create`, and save `TransactionParams` for optional stuff.

Reply:
Yeah I want to change this.  I was thinking I probably need to create some other type to store the transaction info as we build the transaction, then do as you said have create take the optional params as `TransactionParams` and the required params directly.